felix/dataplane/driver.go

```diff
@@ -183,8 +186,19 @@ func StartDataplaneDriver(configParams *config.Config,
 			log.WithError(err).Warning("Unable to assign table index for IPv6 wireguard")
 		}
 
+		// Extract node labels from the hosts such they could be referenced later
+		// e.g. Topology Aware Hints.
+		felixHostname := configParams.FelixHostname
+		felixNode, err := k8sClientSet.CoreV1().Nodes().Get(context.Background(), felixHostname, v1.GetOptions{})
+		if err != nil {
+			log.WithFields(log.Fields{
+				"felixHostname": felixHostname,
+			}).Info("Unabled to extract node labels from Felix host")
+		}
+
 		dpConfig := intdataplane.Config{
-			Hostname:           configParams.FelixHostname,
+			Hostname:           felixHostname,
+			NodeLabels:         felixNode.Labels,
```

Comment:
I think this is ok for the topology lbels, they are unlikely to change, but we need to watch the labels in general. This should go in with a comment with a warning at least!

Reply:
Another option is to only pass in the zone which is constant :crossed_fingers: and nobody gets hurt later.

Reply:
Thanks @tomastigera - as discussed will only pass in the zone which is constant